core/types/block.go

```diff
@@ -1098,7 +1243,7 @@ func (bb *Block) DecodeRLP(s *rlp.Stream) error {
 		return err
 	}
 
-	return s.ListEnd()
+	return nil
```

Comment:
This change looks like a mistake. We should end not only the list of withdrawals, but the main body list as well.

Reply:
Maybe this is an issue with how I'm encoding as RLP, I've encoded the 2 ints and then the 2 lists, there isn't a container list for all 4 properties on the struct.

Reply:
Yeah, you need to call `EncodeStructSizePrefix(payloadSize, ...)` first. Use `(rb RawBody) EncodeRLP(w io.Writer)` as an example.

Reply:
The last line of `func (bb *Block) DecodeRLP` should stay `return s.ListEnd()`